src/pages/emails.ts

```diff
@@ -1,7 +1,12 @@
 import { type APIRoute } from "astro";
-import { emails } from "./index.ts";
+import { emails, emailsByRecipient } from "./index.ts";
 
 export const GET: APIRoute = async ({ request }) => {
+  const url = new URL(request.url)
+  if (url.searchParams.has('recipient')) {
+    return new Response(JSON.stringify(emailsByRecipient[url.searchParams.get('recipient')] || []));
```

Comment:
Let's use `emails.filter()` instead of storing `emailsByRecipient`.

I am using this tool to build a newsletter platform where a significantly large number of emails are sent through the tool. And since it is all stored in memory, this will cause the application to slow down or crash due to large dataset stored in memory.